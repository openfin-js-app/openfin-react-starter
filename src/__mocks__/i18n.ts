import i18n from 'i18next';

i18n
    .init({
        ns: ['landing','config', "menu", ],

        defaultNS: 'landing',
        lng:'en-US',
        fallbackLng:'en-US',

        debug: true,

        interpolation:{
            escapeValue:false,
        }

    });

export default i18n;
