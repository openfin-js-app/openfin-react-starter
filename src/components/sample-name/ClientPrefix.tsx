import * as React from 'react';
import { useContext } from 'react';
import { ApplicationContext } from 'react-openfin';
import { RouteItem } from 'react-openfin-mat-impl'

import { useTranslation } from 'react-i18next';

import cx from 'classnames';

import {makeStyles} from "@material-ui/styles";
import Button from '@material-ui/core/Button';

import dashboardRoutes from '../../routes/Dashboard';
import childWindowRoutes from '../../routes/ChildWindow';

import { clientPrefixComp as style } from '../../assets/jss/sample-name';

interface IProps {
    location?:{
        pathname:string,
    }
}

function makeBrand(props:IProps) {
    let name = null;
    if (props.location.pathname.includes('childWindow')){
        childWindowRoutes.forEach((prop:RouteItem):any => {
            if (prop.path === props.location.pathname){
                name = prop.navbarName;
            }
        });
    }else{
        dashboardRoutes.forEach((prop:RouteItem):any => {
            if (prop.path === props.location.pathname){
                name = prop.navbarName;
            }
        });
    }
    return name;
}

declare const window:any;

const useStyles = makeStyles(style);

const ClientPrefix:React.FunctionComponent<IProps> = (
    props
)=>{

    const { location } = props;

    const classes = useStyles({});

    const {
        actions:{
            onNewSnackBar,
        }
    } = useContext(ApplicationContext);

    const { t, i18n } = useTranslation('menu', { useSuspense: false });

    const bannerLabel = t(makeBrand(props));

    let stylClz;

    if (location.pathname.endsWith('view-one')){
        stylClz = classes.success;
    }else if (location.pathname.endsWith('view-two')){
        stylClz = classes.warning;
    }else{
        stylClz = classes.title;
    }

    const handleBandBtnClick = () =>{
        if (location.pathname.endsWith('view-one')){
            onNewSnackBar({
                message:`Banner of ${bannerLabel} clicked`,
                variant:'success'
            })
        }else if (location.pathname.endsWith('view-two')){
            stylClz = classes.warning;
            onNewSnackBar({
                message:`Banner of ${bannerLabel} clicked`,
                variant:'warning'
            })
        }
    }

    return (
        <Button
            href={"#"} className={stylClz}
            onClick = {handleBandBtnClick}
        >
            {t('appName')+"|"+ bannerLabel}
        </Button>
    )
}

export default ClientPrefix;