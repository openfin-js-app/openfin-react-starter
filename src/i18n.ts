import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import detector from 'i18next-browser-languagedetector';
import backend from 'i18next-xhr-backend';

i18n
    .use(detector)
    .use(backend)
    .use(initReactI18next)
    .init({
        ns: ['landing','config', "menu", ],
        backend:{
            loadPath:`${process.env.PUBLIC_URL}/locales/{{lng}}/{{ns}}.json`,
        },
        defaultNS: 'landing',
        lng:'en-US',
        fallbackLng:'en-US',

        interpolation:{
            escapeValue:false,
        }

    });

export default i18n;
