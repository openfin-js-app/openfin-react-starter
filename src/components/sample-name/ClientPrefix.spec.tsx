import * as React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import Button from '@material-ui/core/Button';
import {Provider} from 'react-redux';
import { InitializeReactOpenfin, ReactOpenfin } from 'react-openfin';
import configurestore from 'redux-mock-store';

import i18n from "../../i18n";
import hist from "../../utils/history";

import ClientPrefix from './ClientPrefix';

declare const jsdom:any;

const mockStore = configurestore();
const initialState = {};

const store = mockStore(initialState);

describe('ClientPrefix comp',()=>{

    let shallow;
    let mount;

    beforeAll(() => {
        InitializeReactOpenfin({
            finUuid: process.env.REACT_APP_FIN_UUID,
            finMockupForceSilentMode:true,
            i18n,
            hist,
            configTabs:[],
            launchBarItems:[],
        });
        mount = createMount();
    });

    beforeEach(() => {
        shallow = createShallow();
        mount = createMount();
    });

    afterEach(()=>{
        mount.cleanUp();
    })

    it('dashboard accessibility',()=>{

        const pathname = '/dashboard/accessibility';


        const wrapper = mount(
            <ReactOpenfin>
                <Provider store = {store}>
                    <ClientPrefix
                        location={{
                            pathname,
                        }}
                    />
                </Provider>
            </ReactOpenfin>
        )

        const btn = wrapper.find(Button);

        expect(btn).toHaveLength(1);

        btn.childAt(0).props().onClick();

        expect(store.getActions()).toHaveLength(0);

    })

    it('dashboard view-one',()=>{

        const pathname = '/dashboard/view-one';

        const wrapper = mount(
            <ReactOpenfin>
                <Provider store = {store}>
                    <ClientPrefix
                        location={{
                            pathname,
                        }}
                    />
                </Provider>
            </ReactOpenfin>
        )

        const btn = wrapper.find(Button);

        expect(btn).toHaveLength(1);

        btn.childAt(0).props().onClick();

        expect(store.getActions()).toHaveLength(0);

    })

    it('dashboard view-two',()=>{

        const pathname = '/dashboard/view-two';

        const wrapper = mount(
            <ReactOpenfin>
                <Provider store = {store}>
                    <ClientPrefix
                        location={{
                            pathname,
                        }}
                    />
                </Provider>
            </ReactOpenfin>
        )

        const btn = wrapper.find(Button);

        expect(btn).toHaveLength(1);

        btn.childAt(0).props().onClick();

        expect(store.getActions()).toHaveLength(0);

    })

    it('childWindow view-one',()=>{

        const pathname = '/childWindow/view-one';

        const wrapper = mount(
            <ReactOpenfin>
                <Provider store = {store}>
                    <ClientPrefix
                        location={{
                            pathname,
                        }}
                    />
                </Provider>
            </ReactOpenfin>
        )

        const btn = wrapper.find(Button);

        expect(btn).toHaveLength(1);

        btn.childAt(0).props().onClick();

        expect(store.getActions()).toHaveLength(0);

    })

    it('childWindow view-two',()=>{

        const pathname = '/childWindow/view-two';

        const wrapper = mount(
            <ReactOpenfin>
                <Provider store = {store}>
                    <ClientPrefix
                        location={{
                            pathname,
                        }}
                    />
                </Provider>
            </ReactOpenfin>
        )

        const btn = wrapper.find(Button);

        expect(btn).toHaveLength(1);

        btn.childAt(0).props().onClick();

        expect(store.getActions()).toHaveLength(0);

    })

})