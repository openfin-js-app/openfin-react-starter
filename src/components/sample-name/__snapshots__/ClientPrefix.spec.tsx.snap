// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientPrefix comp childWindow view-one 1`] = `"http://localhost/childwindow/view-one"`;

exports[`ClientPrefix comp childWindow view-two 1`] = `"http://localhost/childwindow/view-two"`;

exports[`ClientPrefix comp dashboard accessibility 1`] = `"http://localhost/dashboard/accessibility"`;

exports[`ClientPrefix comp dashboard view-one 1`] = `"http://localhost/dashboard/view-one"`;

exports[`ClientPrefix comp dashboard view-two 1`] = `"http://localhost/dashboard/view-two"`;
